dot/sync/chain_sync.go

```diff
@@ -587,21 +640,19 @@ func (cs *chainSync) requestMaxBlocksFrom(bestBlockHeader *types.Header, origin
 // head block number, it would leave us in bootstrap mode forever
 // it would be better to have some sort of standard deviation calculation and discard any outliers (#1861)
 func (cs *chainSync) getTarget() (uint, error) {
-	cs.peerViewLock.RLock()
-	defer cs.peerViewLock.RUnlock()
-
+	peerSetLen := cs.peerViewSet.size()
 	// in practice, this shouldn't happen, as we only start the module once we have some peer states
-	if len(cs.peerView) == 0 {
+	if peerSetLen == 0 {
 		return 0, errNoPeerViews
 	}
 
+	numbers := make([]uint, 0, peerSetLen)
 	// we are going to sort the data and remove the outliers then we will return the avg of all the valid elements
-	uintArr := make([]uint, 0, len(cs.peerView))
-	for _, ps := range cs.peerView {
-		uintArr = append(uintArr, ps.number)
+	for _, view := range cs.peerViewSet.values() {
+		numbers = append(numbers, view.number)
 	}
 
-	sum, count := nonOutliersSumCount(uintArr)
+	sum, count := nonOutliersSumCount(numbers)
```

Comment:
We could move the outliers calculations to the new `peerViewSet` struct, what do you think?